halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/v2/KubernetesV2Service.java

```diff
@@ -513,6 +515,23 @@ default String buildAddress(String namespace) {
     return Strings.join(".", getServiceName(), namespace);
   }
 
+  default Map<String, String> buildEnv(ServiceSettings settings) {
+    Map<String, String> m = new HashMap<>();
+    if (!runsOnJvm()) {
+      return m;
+    }
+
+    // Use half the available memory allocated to the container for the JVM heap
+    m.put("JAVA_OPTS", "-XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -XX:MaxRAMFraction=2");
+    String keyStorePath = KubernetesV2ServiceCa.getJvmKeyStorePath(settings);
+    if (keyStorePath == null || keyStorePath.isEmpty()) {
```

Comment:
Slight nit: I'd prefer to invert this conditional, and store the relevant java_opts here. Reason being: it's very easy for someone to decide to add additional env params below line 531, without realizing they'll only be stored if a keystore path is set.

Reply:
Better now?